src/main/java/classify/textfilecode/TextFileParser.java

```diff
@@ -0,0 +1,199 @@
+package classify.textfilecode;
+
+import classify.data.DataReader;
+import classify.student.Student;
+import classify.student.StudentList;
+import classify.student.SubjectGrade;
+import classify.ui.DataUI;
+import classify.ui.UI;
+import classify.user.InputParsing;
+import classify.user.InvalidCharacterException;
+import classify.user.NameNumberMatchException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+//@@author ParthGandhiNUS
+public class TextFileParser {
+    private static final String MAIN_REGEX = "~~";
+    private static final String SUBJECT_INFO_REGEX = "##";
+    private static final String FILE_SELECTION_PROMPT = "Please enter the exact name of the file " +
+            "you'd like to process:";
+    private static final String REQUEST_TO_TRY_AGAIN = "Please try again!";
+    private static final String TEXT_TYPE = "txt";
+    private static final Character DOT = '.';
+    private static final String NO_MATCH_FOUND = "No Match Found!";
+    private static final String ERROR_ACCESSING_THE_FILE = "Something went wrong while accessing the file";
+    private static final Integer NAME = 0;
+    private static final Integer GENDER = 1;
+    private static final Integer PHONE_NUMBER = 2;
+    private static final Integer LAST_PAYMENT_DATE = 3;
+    private static final Integer REMARKS = 4;
+    private static final Integer SUBJECTS = 5;
+    private static final Integer GRADE_FOR_SUBJECT = 0;
+    private static final Integer NUMBER_OF_CLASSES_ATTENDED = 1;
+    private static final String INPUT_TEXT_FILE_DIRECTORY = "./data/inputFolder";
```

Comment:
Repeated constants from the data package. Perhaps just make them public in their other package and call them over?

Reply:
Second this, superficial point but can also consider implementing printing methods in UI class to reuse within data packages